### PR TITLE
vmauth&vmuser: sync config fields from upstream

### PR DESCRIPTION
**Pull request by Haleygo**
address https://github.com/VictoriaMetrics/operator/issues/939

todo: add more check on vmuser webhook, in case of fields confliction on default route

### CODE REVIEW

File: api/victoriametrics/v1beta1/vmauth_types.go [hunk 1]
```diff
@@ -220,20 +221,87 @@ type VMAuthSpec struct {
...
-	// URLs defines url_prefix for dst routing
+	DropSrcPathPrefixParts *int `json:"drop_src_path_prefix_parts,omitempty"`
+}
+
+type UserConfigOption struct {
+	// DefaultURLs backend url for non-matching paths filter
+	// usually used for default backend with error message
+	DefaultURLs []string `json:"default_url,omitempty"`
+
+	TLSCAFile     string `json:"tls_ca_file,omitempty"`
```
Comment (f41gh7) [inline api/victoriametrics/v1beta1/vmauth_types.go]: For now, I suggest to remove tls related flags and implement it as a separate PR later.

 It must support secret references as scrape components do.

**Review by f41gh7 — COMMENTED** [review]
Looks good, please address comment

File: api/victoriametrics/v1beta1/vmauth_types.go (hunk 1 above)
Comment (Haleygo) [reply api/victoriametrics/v1beta1/vmauth_types.go]: @f41gh7 please check https://github.com/VictoriaMetrics/operator/pull/951/commits/d8eb187245c5d853f53b0b3bed6cdcb8d2b69c56

**Review by f41gh7 — APPROVED** [review]
LGTM

**Pull request merged**